### PR TITLE
Mo changelog mo problem

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
None

#### Purpose of change
Update the changelog for the month of December

#### Describe the solution
tools/generate_changelog.py -f --by-date changelog_12  -t ../repo_token -e 2020-12-01 2020-11-01
tac changelog_12 > changelog_12_rev
tools/merge_summaries.sh changelog_12_rev data/changelog.txt
Then copyedit the results.

#### Additional context
Typos happen, and some of the things I prune might be bigger than I remember or part of some larger set of changes that should go in the changelog, please take a look at not just the final diff, but also the copyedit commit to check for missed entries.

### CODE REVIEW

File: data/changelog.txt [hunk 1]
```diff
@@ -87,6 +95,9 @@ Display contained items like 'water (plastic bottle)' instead of 'plastic bottle
 Persist preferred aiming mode.
 Makes Chibi Ultica a stand-alone tileset.
 Brings back MSX+.
+Add item length and pocket volume entries to menus.
+Add PAGE_UP/DOWN-fast-scroll to many menus with long lists.
+NewCharacter - Add responsive screen behavior for small terminal sizes.
```
Comment (ZhilkinSerg) [inline data/changelog.txt:100]:
```suggestion
Add responsive screen behavior to new character menu for small terminal sizes.
```

File: data/changelog.txt [hunk 2]
```diff
@@ -87,6 +95,9 @@ Display contained items like 'water (plastic bottle)' instead of 'plastic bottle
 Persist preferred aiming mode.
 Makes Chibi Ultica a stand-alone tileset.
 Brings back MSX+.
+Add item length and pocket volume entries to menus.
+Add PAGE_UP/DOWN-fast-scroll to many menus with long lists.
```
Comment (ZhilkinSerg) [inline data/changelog.txt:99]:
```suggestion
Adds fast scroll to many menus with long lists.
```

File: data/changelog.txt [hunk 3]
```diff
@@ -209,6 +222,11 @@ Improves rendering speed by optimizing find_tile_looks_like method (FPS improved
 Greatly reduce overhead of processing idle vehicles.
 Optimizes Creature::has_effect (performance improved ~9.5%)
 Adds a specialization for static/interned string_ids  (performance improved ~30%))
+Cache translations and _() macro (LOCALIZE=1 build perf improved ≈20%)
+Major optimization in field processing.
+Add reachability cache/filter that can reject bresenham-based LOS checks in O(1) time.
+Use field_cache in map::get_field to speedup fire check in npc::assess_danger.
+Speedup quicksave (mapbuffer::save) by limiting `ui_manager::redraw()` rate
```
Comment (ZhilkinSerg) [inline data/changelog.txt:229]:
```suggestion
Speedup quicksave by limiting ui redraw rate
```

File: data/changelog.txt [hunk 4]
```diff
@@ -209,6 +222,11 @@ Improves rendering speed by optimizing find_tile_looks_like method (FPS improved
 Greatly reduce overhead of processing idle vehicles.
 Optimizes Creature::has_effect (performance improved ~9.5%)
 Adds a specialization for static/interned string_ids  (performance improved ~30%))
+Cache translations and _() macro (LOCALIZE=1 build perf improved ≈20%)
```
Comment (ZhilkinSerg) [inline data/changelog.txt:225]:
```suggestion
Improve performance of builds with enabled localization by ≈20% by caching translations.
```

File: data/changelog.txt [hunk 5]
```diff
@@ -61,14 +62,21 @@ Adds 'What the hell happened' stories to NPCs, keyed off backgrounds.
...
 Added many proficiencies and uses for them.
-Add frog mother monster.
+Adds frog mother monster.
 Adds mega mutant monsters.
+Adds a selection of mutated woodland creatures.
+New 'phase skulker' zombies teleport and haunt between the hallways of microlabs.
+Make mutant organs and lungs contain toxins.
+Adds Streetlights to sidewalks; traffic lights, stop lines and pedestrian crossings to intersections.
+Adds Rust Zombies to the Game.
+Diversify venom sources and effects.
```
Comment (actual-nh) [inline data/changelog.txt:78]:
```suggestion
Adds streetlights to sidewalks, and traffic lights, stop lines, and pedestrian crossings to intersections.
Adds Rust Zombies.
Diversifies venom sources and effects.
```

File: data/changelog.txt (hunk 1 above)
Comment (actual-nh) [reply data/changelog.txt:100]:
```suggestion
Adds responsive screen behavior for small terminal sizes to new character menu.
```

File: data/changelog.txt [hunk 6]
```diff
@@ -195,6 +206,8 @@ Picking up an item stack fills up your pockets if it doesn't fit into one pocket
 Wielding combines item stacks if possible.
 Robots can no longer be disabled through melee hits.
 Fixes auto-consume zones in vehicles.
+Fix monster special attack range limiting code.
```
Comment (actual-nh) [inline data/changelog.txt:209]:
```suggestion
Fixes monster special attack range-limiting code.
```

File: data/changelog.txt (hunk 4 above)
Comment (actual-nh) [reply data/changelog.txt:225]:
```suggestion
Improves performance of builds with localization enabled by ~20% via caching translations.
```

File: data/changelog.txt [hunk 7]
```diff
@@ -43,6 +43,7 @@ Artifacts regenerate ammo option.
 Added ACTIVE enchantment condition.
 Allow recipes to specify individual activity levels.
 Add weariness, a sort of long term stamina that limits the character's ability to be extremely active 24/7.
+Report faction camp build sites and improve tutorial.
```
Comment (actual-nh) [inline data/changelog.txt:46]:
```suggestion
Adds weariness, a sort of long-term stamina that limits characters' ability to be extremely active 24/7.
Report faction camp build sites and improve faction camp tutorial.
```
Am I correct re "faction camp" regarding the tutorial? (I've never built one.)

File: data/changelog.txt [hunk 8]
```diff
@@ -61,14 +62,21 @@ Adds 'What the hell happened' stories to NPCs, keyed off backgrounds.
 Feral Humans with blob psychosis.
 Allow plank and branch bundles to be made from cordage, vines, or wire.
 Microlabs can contain artifacts in containment.
-Add NPC_class and NPC json templates and instructions.
+Adds NPC_class and NPC json templates and instructions.
 Adapts almost all vehicle definitions to use directional parts (most of the work by Curstwist).
```
Comment (actual-nh) [inline data/changelog.txt:66]:
```suggestion
Adapts almost all vehicle definitions to use directional parts (most work done by Curstwist).
```

**Review by actual-nh — COMMENTED** [review]
Again, what is desirable re verb tenses? @ToxiClay, any thoughts?

**Comment by ToxiClay** [comment]
> Again, what is desirable re verb tenses? @ToxiClay, any thoughts?

In my opinion, the plural form is better. The changelog is describing what each individual change does, and just like in commit messages, you'd say something to the effect of "[This change] adds weariness, a sort of long-term stamina...."

The singular form reads more like an imperative, like you'd find in a `TODO` comment.

**Comment by actual-nh** [comment]
> In my opinion, the plural form is better. The changelog is describing what each individual change does, and just like in commit messages, you'd say something to the effect of "[This change] adds weariness, a sort of long-term stamina...."

That's what I thought. Another way to look at it, in this case, would be "[Version 0.F] adds weariness, a sort of long-term stamina...".

> The singular form reads more like an imperative, like you'd find in a `TODO` comment.

Good point.

File: data/changelog.txt (hunk 5 above)
Comment (KorGgenT) [reply data/changelog.txt:78]: keep the semicolon otherwise the sentence structure's readability dips

File: data/changelog.txt (hunk 7 above)
Comment (kevingranade) [reply data/changelog.txt:46]: I was actually puzzeled by that too, but now that you mention it, that's what it is.

File: data/changelog.txt [hunk 9]
```diff
@@ -1,4 +1,4 @@
-# 0.F (2020-11-01)
+# 0.F (2020-12-01)
 
 ## Features:
 Added option to randomize INITIAL_DAY.
```
Comment (actual-nh) [inline data/changelog.txt:4]:
```suggestion
Adds option to randomize INITIAL_DAY.
```

File: data/changelog.txt [hunk 10]
```diff
@@ -43,6 +43,7 @@ Artifacts regenerate ammo option.
 Added ACTIVE enchantment condition.
 Allow recipes to specify individual activity levels.
 Add weariness, a sort of long term stamina that limits the character's ability to be extremely active 24/7.
```
Comment (actual-nh) [inline data/changelog.txt:45]:
```suggestion
Adds ACTIVE enchantment condition.
Allows recipes to specify individual activity levels.
Adds weariness, a sort of long term stamina that limits the character's ability to be extremely active 24/7.
```

File: data/changelog.txt [hunk 11]
```diff
@@ -61,14 +62,21 @@ Adds 'What the hell happened' stories to NPCs, keyed off backgrounds.
 Feral Humans with blob psychosis.
 Allow plank and branch bundles to be made from cordage, vines, or wire.
```
Comment (actual-nh) [inline data/changelog.txt:63]: BTW, regarding feral humans, I've been trying to figure out how best to have some of the more-sapient variety. Getting them to use guns appropriately looks to be a difficulty.
```suggestion
Allows plank and branch bundles to be made from cordage, vines, or wire.
```

File: data/changelog.txt [hunk 12]
```diff
@@ -195,6 +206,8 @@ Picking up an item stack fills up your pockets if it doesn't fit into one pocket
 Wielding combines item stacks if possible.
 Robots can no longer be disabled through melee hits.
 Fixes auto-consume zones in vehicles.
+Fix monster special attack range limiting code.
+Falling vehicles: prevent bouncing after the first collision with the ground.
 
 ## Performance:
 Add option to enable batching when it is available in SDL.
```
Comment (actual-nh) [inline data/changelog.txt:213]:
```suggestion
Falling vehicles: prevents bouncing after the first collision with the ground.

## Performance:
Adds an option to enable batching when it is available in SDL.
```

File: data/changelog.txt [hunk 13]
```diff
@@ -209,6 +222,11 @@ Improves rendering speed by optimizing find_tile_looks_like method (FPS improved
 Greatly reduce overhead of processing idle vehicles.
```
Comment (actual-nh) [inline data/changelog.txt:222]:
```suggestion
Greatly reduces overhead from processing idle vehicles.
```

File: data/changelog.txt [hunk 14]
```diff
@@ -209,6 +222,11 @@ Improves rendering speed by optimizing find_tile_looks_like method (FPS improved
...
 Optimizes Creature::has_effect (performance improved ~9.5%)
 Adds a specialization for static/interned string_ids  (performance improved ~30%))
+Cache translations and _() macro (LOCALIZE=1 build perf improved ≈20%)
+Major optimization in field processing.
+Add reachability cache/filter that can reject bresenham-based LOS checks in O(1) time.
+Use field_cache in map::get_field to speedup fire check in npc::assess_danger.
```
Comment (actual-nh) [inline data/changelog.txt:228]:
```suggestion
Major optimization of field processing.
Adds reachability cache/filter that can reject bresenham-based LOS checks in O(1) time.
Uses field_cache in map::get_field to speedup fire check in npc::assess_danger.
```

File: data/changelog.txt [hunk 15]
```diff
@@ -209,6 +222,11 @@ Improves rendering speed by optimizing find_tile_looks_like method (FPS improved
...
+Speedup quicksave (mapbuffer::save) by limiting `ui_manager::redraw()` rate
 
 ## Infrastructure:
 Add basic infrastructure for diseases.
```
Comment (actual-nh) [inline data/changelog.txt:232]:
```suggestion
Adds basic infrastructure for diseases.
```

**Review by actual-nh — COMMENTED** [review]
A few more - mostly verb tense, but a few other fixes/improvements.

File: data/changelog.txt (hunk 8 above)
Comment (kevingranade) [inline data/changelog.txt:66]:
```suggestion
Adapts almost all vehicle definitions to use directional parts.
```

File: data/changelog.txt (hunk 7 above)
Comment (kevingranade) [inline data/changelog.txt:46]:
```suggestion
Adds ACTIVE enchantment condition.
Allows recipes to specify individual activity levels.
Adds weariness, a sort of long term stamina that limits characters' ability to be extremely active 24/7.
Reports faction camp build sites and improves faction camp tutorial.
```

**Pull request merged**